epysurv/simulation/seasonal_noise.py

```diff
@@ -0,0 +1,69 @@
+import rpy2.robjects.packages as rpackages
+import pandas as pd
+
+from rpy2 import robjects
+
+from epysurv.simulation.utils import r_list_to_frame, add_date_time_index_to_frame
+
+
+surveillance = rpackages.importr("surveillance")
+
+
+def simulate_outbreaks(
+    length: int,
+    amplitude=1,
+    alpha=1,
+    beta=0,
+    phi=0,
+    frequency=1,
+    state=None,
+    state_weight=0,
+) -> pd.DataFrame:
+    """Generation of a cyclic model of a Poisson distribution as background data for a simulated timevector.
+
+    The mean of the Poisson distribution is modelled as:
+    :math:`\\mu = \\exp{(A\\sin{(frequency \\cdot (t + \\phi))} + \\alpha + \\beta * t + L * state)}`
+
+    Attributes
+    ----------
+    amplitude
+        amplitude (range of sinus), default = 1.
+    alpha
+        parameter to move along the y-axis (negative values not allowed) with alpha >= amplitude, default = 1.
+    beta
+        regression coefficient, default = 0.
+    phi
+        factor to create seasonal moves (moves the curve along the x-axis), default = 0.
+    length
+        number of weeks to model.
+    frequency
+        factor to determine the oscillation-frequency, default = 1.
+    state
+        if a state chain is entered the outbreaks will be additional weighted by state_weight
+    state_weight
+        additional weight for an outbreak which influences the distribution parameter mu, default = 0.
+
+    Returns
+    -------
+    A DataFrame of an endemic time series that contains n weeks where n=``length``.
+    The DataFrame is divided into timesteps where each step is equivalent to one calender week.
+    It contains a ``mean`` column which is the mean case count according to the sinus based model.
+    And finally, it contains a column ``n_cases`` that consists of the generates case counts
+    based on the sinus model
+    """
+    simulated = surveillance.sim_seasonalNoise(
+        A=amplitude,
+        alpha=alpha,
+        beta=beta,
+        phi=phi,
+        length=length,
+        frequency=frequency,
+        state=robjects.NULL if state is None else robjects.IntVector(state),
```

Comment:
This won't work, as it relies on a variable named `state` being set outside the function. Just set the default to `None` and do the conversion inside the function.

Reply:
But isn't `state` always set to `None` since the default parameter of `simulate_outbreaks()` is set to `None` for `state`? Also, what do you mean inside the function? The function I am calling here is the R function itself. Thanks for the clarification :dancing_men: